mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionMetadataTypeUtils.java

```diff
@@ -346,8 +352,14 @@ public static boolean areTypesEqual(MetadataType type, MetadataType otherType, C
 
     if (typeClassName != null && otherTypeClassName != null) {
       if (classLoader != null) {
-        if (!areAssignable(typeClassName, otherTypeClassName, classLoader)) {
-          return false;
+        try {
+          if (!areAssignable(typeClassName, otherTypeClassName, classLoader)) {
+            return false;
+          }
+        } catch (ClassNotFoundException e) {
+          LOGGER.debug(format("Classes %s and %s are not available in the class loader for comparison", typeClassName,
```

Comment:
What's the point of hiding the exception here? If you cannot load the class, the problem is going to pop up further up the road anyway.

Reply:
what do you mean the problem is going to pop up later? This is a best effort approach. If we have the classes we try to use them and see if they are assignable, otherwise we try to determine if they are equal through other approaches (which is the case when running with the AST). See https://github.com/mulesoft/mule/pull/9002 for why this code was done in the first place.

Reply:
But this is not checking `if we have classes`. By catching `ClassNotFoundException` you're hiding the case in which there is a class but you cannot load it, most likely because of classloading isolation. Therefore you're most likely hiding an extension with API definition issues. Also, bad usage of LOGGER.debug. Either use Slf4j interpolation or add an `isDebugEnabled()` condition